returnn/dataset.py

```diff
@@ -149,3 +149,110 @@ def run(self):
                 hdf_writer.insert_batch(numpy.asarray([[speaker_index]], dtype="int32"), [1], [segment_name])
 
         hdf_writer.close()
+
+
+class ExtractSeqLensJob(Job):
+    """
+    Extracts sequence lengths from a dataset for one specific key.
+
+    Also see: :class:`i6_experiments.users.schmitt.corpus.statistics.GetSeqLenFileJob`
+    """
+
+    # noinspection PyShadowingBuiltins
+    def __init__(
+        self,
+        dataset: Dict[str, Any],
+        post_dataset: Optional[Dict[str, Any]] = None,
+        *,
+        key: str,
+        format: str,
+        returnn_config: Optional[ReturnnConfig] = None,
+    ):
+        """
+        :param dataset: dict for :func:`returnn.datasets.init_dataset`
+        :param post_dataset: extension of the dataset dict, which is not hashed
+        :param key: e.g. "data", "classes" or whatever the dataset provides
+        :param format: "py" or "txt"
```

Comment:
maybe output_format is a more descriptive name? At least I did not understand what this is about without looking at the actual code.

Also maybe a better explanation what py or txt means here would be helpful, but this is optional. I guess most people that want to use this job know what it looks like from working with other jobs. I was just surprised that for txt it does store the seq name, while for py it does.